### PR TITLE
Issue/5543 resource file upload should ignore identical files

### LINKED ISSUE #5543: resource file upload should ignore identical files
The current bulk resource file upload may fail when it gets the same file contents more than once in the same request. Because of #2313 this may happen in practice (see its first comment). Identical files should instead be ignored, they are identical after all.

Fix this in the `stat_files` endpoint.

### PR DESCRIPTION
**Pull request by Hugo-Inmanta**
# Description

The stat_file API endpoint now returns a set of files instead of a list.

closes #5543  

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [x] Attached issue to pull request
- [x] Changelog entry
- [x] Type annotations are present
- [x] Code is clear and sufficiently documented
- [x] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )

### CODE REVIEW

File: changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+description: The stat_file API endpoint now returns a list with no duplicates.
+change-type: patch
+destination-branches:
+  - master
+  - iso6
+  - iso5
+sections:
+  bugfix: "{{description}}"
```
Comment (Hugo-Inmanta) [inline changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml:8]: I don't have a strong opinion about this changelog entry being present or not

File: src/inmanta/server/services/fileservice.py [hunk 2]
```diff
@@ -131,13 +131,14 @@ def stat_file_internal(self, files: Iterable[str]) -> List[str]:
         """
         Return which files in the list don't exist on the server
         """
-        response: List[str] = []
+        response: Dict[str, object] = {}
```
Comment (Hugo-Inmanta) [inline src/inmanta/server/services/fileservice.py:134]: I decided to use a dict here where only the keys are of interest to have an efficient set-like behavior and preserving insertion order compared to the previous implementation.
I'm not sure whether this is truly necessary or not

File: changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml (hunk 1 above)
Comment (sanderr) [reply changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml:8]: I think the message is good as an internal description of the change. For the end-user message I'd focus more on the symptom: this fixes a bug that may cause the first export for an environment to fail if there are identical files.

File: src/inmanta/server/services/fileservice.py (hunk 2 above)
Comment (sanderr) [reply src/inmanta/server/services/fileservice.py:134]: Looks good to me. Perhaps add a comment.

Comment (sanderr) [inline src/inmanta/server/services/fileservice.py:134]:
```suggestion
        response: dict[str, object] = {}
```

**Review by sanderr — APPROVED** [review]

**Review by sanderr — CHANGES_REQUESTED** [review]
Forgot that I requested a change. The one to the change entry, the rest are minor.

File: changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml (hunk 1 above)
Comment (Hugo-Inmanta) [reply changelogs/unreleased/5543-resource-file-upload-should-ignore-identical-files.yml:8]: Great point :+1:

**Review by sanderr — APPROVED** [review]

**Comment by sanderr** [comment]
@bartv FYI

**Pull request closed without merging**